metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -820,21 +820,7 @@ class MetalsLanguageServer(
       params: ReferenceParams
   ): CompletableFuture[util.List[Location]] =
     CancelTokens { _ =>
-      val timer = new Timer(time)
-      val result = referencesResult(params)
-      if (config.statistics.isReferences) {
-        if (result.symbol.isEmpty) {
-          scribe.info(s"time: found 0 references in $timer")
-        } else {
-          scribe.info(
-            s"time: found ${result.locations.length} references to symbol '${result.symbol}' in $timer"
-          )
-        }
-      }
-      if (result.symbol.nonEmpty) {
-        compileAndLookForNewReferences(params, result)
-      }
-      result.locations.asJava
+      referencesResult(params).locations.asJava
```

Comment:
Just moved the implementation into `referencesResult` that returns `ReferencesResult` instead of `CompletableFuture[util.List[Location]]`.